### ISSUE TITLE
Power operator does not work with floating point in -betterC

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/dmd#20201]
### CC List

* Iain Buclaw (@ibuclaw)
* Nick Treleaven (@ntrel)

### Description

````markdown
As of dmd 2.101.0 power operator doesn't work in -betterC for double/double.

Due to using std.math : pow as the hook for this expression, which depends upon exp2 which isn't templated.


```
extern(C) void main() {
    double input = 3f;
    double power = 2;
    double calculated = input ^^ power;
}
```

```
verify_pow_reg.obj : error LNK2019: unresolved external symbol __D3std4math11exponential4exp2FNaNbNiNeeZe referenced in function __D3std4math11exponential__T3powTdTdZQjFNaNbNiNeddZ4implFNaNbNiNfeeZe
```
````

**Comment by dlangBugzillaToGithub** [comment]
#### nick (@ntrel) commented on 2022-12-04T17:57:59Z



````markdown
Each exp2 overload forwards to template exp2Impl, and each are marked pragma(inline, true). So this should work if the inlining is done. (Though currently it makes no difference using -inline -O or not).
````

**Comment by dlangBugzillaToGithub** [comment]
#### alphaglosined commented on 2022-12-04T22:00:25Z



````markdown
Unfortunately, run.dlang.io's run-all D compiler option appears to be broken, so I can't test if the recent changes to dmd's inliner caused this to break and hence be a regression.

But yeah I see three possibilities here:

1) Fix inlining
2) Template exp functions
3) Use libc pow instead

Of course, all these options are ignoring the forest. Why are there compiler hooks implemented in Phobos?
````

**Comment by dlangBugzillaToGithub** [comment]
#### ibuclaw (@ibuclaw) commented on 2023-02-07T00:30:01Z



````markdown
Any `T ^^ float` operation is by definition non-trivial, so I'm tempted to just close this as wontfix.
````

**Comment by dkorpel** [comment]
Let's keep progress of this issue in https://github.com/dlang/dmd/issues/22670
